### ISSUE TITLE
Menu is inconsitent, cache isn't getting invalidated.

### ISSUE THREAD

**Issue by bxm156** [divio/django-cms#5068]
DjangoCMS == 3.2.1

On my site, when I load the page, sometimes my menu bar only shows the Homepage in the menu, other times, it shows what I would expect, which is the Homepage + any recently created/published pages.

I suspect its maybe a caching issue? Something isn't getting invalidated. I have Nginx + 8 uwsgi threads. Perhaps some of these other UWGI processors are using old cache data, thus creating this affect not sometimes not seeing new pages in my menu.

However, I disabled caching as best as I could...
```
CMS_PAGE_CACHE = False
CMS_PLACEHOLDER_CACHE = False
CMS_PLUGIN_CACHE = False
```

(I'm using the default Django File Cache Bakend atm. I'm looking to move to something better, but right now on a single server this shouldn't make a difference)

Since my nginx and django run in a docker container. I tried restarting the container, which would blow away any cache. As expected, once restarted, the menu consistently shows all the pages I would expect.

**Comment by bxm156** [comment]
Note: I don't have FetchFromCacheMiddleware and UpdateCacheMiddleware in my settings.py, as it wasn't included in the How To Setup documentation. (And I thought I turned of caching, but it would seem thats not the case).

By default, Django uses a file cache, so maybe those should be "required" for everyone? Its hard to tell the settings I listed above have really turned of caching, as the documentation may be out of date:
https://github.com/divio/django-cms/issues/3961

**Comment by czpython** [comment]
@bxm156 
There was a bug on the menus cache setting, was fixed in 8fbd4fd0a6ff640febd1f12e7ea99df52fafec8f
To turn off menu cache you'll need to set ``CMS_CACHE_DURATIONS``.

Regarding cache invalidation, this is sadly a known issue which we're aiming to fix with the menus cleanup/refactor.
